.github/workflows/pull_request.yml

```diff
@@ -59,14 +81,18 @@ jobs:
           name: Android ${{ env.ARCHIVE_KEY }} Firebase Debug Log
           path: "**/firebase-debug.log"
   build-js:
+    needs: jobMatrixSetup
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.jobMatrixSetup.outputs.js_test_jobs_matrix) }}
     steps:
       - uses: actions/checkout@v4
       - name: Setup test environment
         uses: ./.github/actions/setup_test_action
         timeout-minutes: 10
       - name: Run JS Tests
-        run: ./gradlew cleanTest jsTest
+        run: ./gradlew ${{ matrix.gradle_tasks }}
       - name: Upload JS test artifact
         uses: actions/upload-artifact@v4
```

Comment:
All artifacts should have a unique name, so it should derive the name of the artifact from the gradle task. E.g "JS firebase-firestore Firebase Debug Log"